### PR TITLE
doc: Added Linked CI with Child Info + Runtime Build Parameters

### PR DESCRIPTION
**Pull request by ashokdevtron**
Added the following to our existing docs:
- Passing Build Parameter section in Application & Application Group docs
- Modified Linked CI section to include Child Info
- RBAC and snapshots included

### CODE REVIEW

File: docs/user-guide/application-groups.md [hunk 1]
```diff
@@ -53,11 +53,17 @@ The **Build & Deploy** tab of your application group enables you to trigger the
 
     ![Figure 5: Selecting Commit](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/application-groups/select-commit-1.jpg)
 
-    Repeat the step for other applications, and then click **Start Build**.
+{% hint style="info" %}
+### Passing Build Parameters [![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/elements/EnterpriseTag.svg)](https://devtron.ai/pricing)
+
+Similar to application, you can also [pass build parameters](./deploying-application/triggering-ci.md#passing-build-parameters) in application groups before triggering the build.
```
Comment (abhibhaw) [inline docs/user-guide/application-groups.md:59]: Screenshot missing how to do?

File: docs/user-guide/creating-application/workflow/ci-pipeline.md [hunk 2]
```diff
@@ -210,18 +210,25 @@ If one code is shared across multiple applications, `Linked Build Pipeline` can
 2. On the **App Configuration** page, select **Workflow Editor**.
 3. Select **+ New Workflow**.
 4. Select **Linked Build Pipeline**.
-5. Enter the following fields on the **Create linked build pipeline** screen:
+5. On the **Create linked build pipeline** screen:
+    * Search for the application in which the source CI pipeline is present.
+    * Select the source CI pipeline from the application that you selected above.
+    * Enter a new name for the linked CI pipeline. It cannot be the same as the source CI pipeline.
```
Comment (abhibhaw) [inline docs/user-guide/creating-application/workflow/ci-pipeline.md:216]: `It cannot be the same as the source CI pipeline.` It has similar validation as other pipelines or something different?

File: docs/user-guide/deploying-application/triggering-ci.md [hunk 3]
```diff
@@ -22,13 +22,51 @@ CI builds can be time-consuming for large repositories, especially for enterpris
...
+---
+
+## Passing Build Parameters [![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/elements/EnterpriseTag.svg)](https://devtron.ai/pricing)
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need to have [Build & deploy permission](../global-configurations/authorization/user-access.md#role-based-access-levels) or above (along with access to the environment and application) to pass build parameters.
+{% endhint %}
+
+If you wish to pass runtime parameters for build job, you can provide key-value pairs before triggering the build. This will inject those key-value pairs as environment variables in CI runner pods and all its containers.
```
Comment (abhibhaw) [inline docs/user-guide/deploying-application/triggering-ci.md:40]: Can we try re-writing this in generic terms not how we do it.. Maybe like you can access the passed values with keys from your environment variables...

File: docs/user-guide/deploying-application/triggering-ci.md [hunk 4]
```diff
@@ -22,13 +22,51 @@ CI builds can be time-consuming for large repositories, especially for enterpris
...
+3. Enter your key-value pair as shown below. 
+
+    ![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/triggering-ci/key-value.jpg)
+
+    <br /> Similarly, you may add more than one key-value pair by using the **+ Add Parameter** button.
+
+4. Click **Start Build**.
+
+{% hint style="info" %}
+Passing build parameters is currently not supported for [Linked Build pipeline](../creating-application/workflow/ci-pipeline.md#2-linked-build-pipeline) and [External CI pipeline](../creating-application/workflow/ci-pipeline.md#3-deploy-image-from-external-service).
```
Comment (abhibhaw) [inline docs/user-guide/deploying-application/triggering-ci.md:61]: This is irrelevant... Linked CI and external CI doesn't have a build step so build parameter doesn't make sense

**Review by abhibhaw — CHANGES_REQUESTED** [review]

File: docs/user-guide/application-groups.md (hunk 1 above)
Comment (ashokdevtron) [reply docs/user-guide/application-groups.md:59]: Added snapshots here as well

File: docs/user-guide/creating-application/workflow/ci-pipeline.md (hunk 2 above)
Comment (ashokdevtron) [reply docs/user-guide/creating-application/workflow/ci-pipeline.md:216]: Okay. Removing that line.

File: docs/user-guide/deploying-application/triggering-ci.md (hunk 3 above)
Comment (ashokdevtron) [reply docs/user-guide/deploying-application/triggering-ci.md:40]: Done

**Review by abhibhaw — APPROVED** [review]
lgtm

**Review by uxarya-d — APPROVED** [review]

**Pull request merged**